pkg/lib/oauth/handler/handler_authz.go

```diff
@@ -400,17 +345,43 @@ func (h *AuthorizationHandler) prepareConsentRequest(ctx context.Context, req *h
 	}, nil
 }
 
+func (h *AuthorizationHandler) HandleRequestWithTx(
```

Comment:
Context should be the first argument. Never embed it in an options struct.

Reply:
I wanted to enforce the Context used here must be the context returned by ValidateRequestWithoutTx, therefore I put it inside the struct, but this pattern is indeed a bit weird.